src/Domain.LinnApps/Reports/ShortagesReportService.cs

```diff
@@ -0,0 +1,73 @@
+﻿namespace Linn.Purchasing.Domain.LinnApps.Reports
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    using Linn.Common.Persistence;
+    using Linn.Common.Reporting.Models;
+    using Linn.Purchasing.Domain.LinnApps.Reports.Models;
+
+    public class ShortagesReportService : IShortagesReportService
+    {
+        private readonly IRepository<ShortagesEntry, string> shortagesEntryView;
```

Comment:
Since this is mapped to a View (not a table) and you've declared it as having .HasNoKey() in the ServiceDbContext, you shouldn't really use IRepopsitory (since you're implying a key type of string here when there is no key). IQueryRepository lets you define a repository without a key type which is more suitable

Reply:
OK, noted!